addons/account/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,176 @@
+                    **{k: v for k, v in sending_settings.items() if k in sending_settings_whitelist},
+                    'mail_partner_ids': scheduled_message.partner_ids.ids,
+                    'mail_subject': scheduled_message.subject,
+                    'mail_body': scheduled_message.body,
+                    'mail_attachments_widget': payload_attachments_widget + manual_attachments_widget_data,
+                }
+
+            # The generate and send invoice logic (including any EDI sending methods) with extra manual attachment if any
+            attachments = self.env['account.move.send']._generate_and_send_invoices(
+                move_id,
+                payload.get('allow_fallback_pdf'),
+                **{k: v for k, v in sending_settings.items() if k in sending_settings_whitelist},
+            )
+            if attachments and notification_parameters.get('from_move_send_wizard') and scheduled_message.is_note:
+                body = scheduled_message.body.replace('scheduled to send', _('sent'))
+                move_id.message_post(
+                    subject=scheduled_message.subject,
+                    body=body,
+                    subtype_xmlid='mail.mt_note',
+                )
+            scheduled_message.unlink()
+
+        if remained_scheduled_messages:
+            super(MailScheduledMessage, remained_scheduled_messages)._post_message(raise_exception)
+
+    # ------------------------------------------------------
+    # Business Methods
+    # ------------------------------------------------------
+
+    @api.model
+    def _sending_settings_whitelist(self):
+        """ Sending Settings that can be used when generating and sending invoices.
+        """
+        return {
+            'sending_methods',
+            'invoice_edi_format',
+            'extra_edis',
+            'pdf_report',
+            'author_user_id',
+            'author_partner_id',
+            'mail_template',
+            'mail_lang',
+            'mail_body',
+            'mail_subject',
+            'mail_partner_ids',
+            'mail_attachments_widget',
+            'allow_fallback_pdf',
+        }
+
+    @api.model
+    def _post_messages_cron(self, limit=50):
+        # OVERRIDE
+        # To set OdooBot as a author of the messages from account.move.send.wizard
+        domain = [('scheduled_date', '<=', fields.Datetime.now())]
+        messages = self.search(domain, limit=limit)
+        messages_from_move = messages.filtered(lambda m: json.loads(m.notification_parameters or '{}').get('from_move_send_wizard'))
+        if messages_from_move:
+            _logger.info("Posting %s scheduled messages", len(messages_from_move))
+            messages_from_move.with_context(mail_notify_force_send=True).with_user(SUPERUSER_ID)._post_message(raise_exception=False)
+
+        super(MailScheduledMessage, messages - messages_from_move)._post_messages_cron(limit=limit)
+
+    def get_sending_method_and_edi_data(self):
```

Comment:
Prefix by module (as it is completely account related) + make private by default
```suggestion
    def _account_get_sending_and_edi_data(self):
```